ci-operator/step-registry/ipi/conf/gcp/user-labels-tags/ipi-conf-gcp-user-labels-tags-commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+PATCH="${SHARED_DIR}/user-labels-tags.yaml.patch"
+cat > "${PATCH}" << EOF
+platform:
+  gcp:
+    userLabels:
+    - key: createdby
+      value: installer-qe
+    - key: environment
+      value: test
+    userTags:
```

Comment:
azure and aws already support userTags feature, I think we should keep the same style like them, e.g: 
./azure/ipi/usertags/cucushift-installer-rehearse-azure-ipi-usertags-workflow.yaml
./aws/ipi/usertags/provision/cucushift-installer-rehearse-aws-ipi-usertags-provision-chain.yaml

cc @yunjiang29 to review too.

Reply:
In terms GCP user tags, those tags must be pre-configured beforehand. Is it true for azure and aws?

Reply:
Good to know, azure and aws sounds not that case.

So the tag has to be on organizational level? If users specify a non-organizational-level tag, installer would thrown pre-check error?

Reply:
> The user tags can be on organizational level, or under project level.

If the specified userTags did not exiting on both org and project level, what would happend?

Reply:
Good question, let me confirm.

Reply:
The installation can succeed, but the cluster operator image-registry would stay processing with error like "failed to add tag(s) to ... bucket". Any specified user tag which has non-existing key or non-existing value won't be applied to image-registry storage bucket. 

FYI currently I only find out way of checking image-registry bucket's user tags via GCP web console, so no post-installation check for user tags yet.

Reply:
Per the definition of `userTags` and `userLables` [1], various character combinations should be covered in CI, and let's configure the maximum number of tags: 50.

[1]
```
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <[]object>
  userLabels has additional keys and values that the installer will add as labels to all resources that it creates on GCP. Resources created by the cluster itself may not include these labels. This is a TechPreview feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags featureGate enabled or TechPreviewNoUpgrade featureSet to configure labels.

FIELDS:
    key <string> -required-
      key is the key part of the label. A label key can have a maximum of 63 characters and cannot be empty. Label must begin with a lowercase letter, and must contain only lowercase letters, numeric characters, and the following special characters `_-`.

    value <string> -required-
      value is the value part of the label. A label value can have a maximum of 63 characters and cannot be empty. Value must contain only lowercase letters, numeric characters, and the following special characters `_-`.

KIND:     InstallConfig
VERSION:  v1

RESOURCE: <[]object>
  userTags has additional keys and values that the installer will add as tags to all resources that it creates on GCP. Resources created by the cluster itself may not include these tags. Tag key and tag value should be the shortnames of the tag key and tag value resource. This is a TechPreview feature and requires setting CustomNoUpgrade featureSet with GCPLabelsTags featureGate enabled or TechPreviewNoUpgrade featureSet to configure tags.

FIELDS:
    key <string> -required-
      key is the key part of the tag. A tag key can have a maximum of 63 characters and cannot be empty. Tag key must begin and end with an alphanumeric character, and must contain only uppercase, lowercase alphanumeric characters, and the following special characters `._-`.

    parentID <string> -required-
      parentID is the ID of the hierarchical resource where the tags are defined, e.g. at the Organization or the Project level. To find the Organization ID or Project ID refer to the following pages: https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id, https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects. An OrganizationID must consist of decimal numbers, and cannot have leading zeroes. A ProjectID must be 6 to 30 characters in length, can only contain lowercase letters, numbers, and hyphens, and must start with a letter, and cannot end with a hyphen.

    value <string> -required-
      value is the value part of the tag. A tag value can have a maximum of 63 characters and cannot be empty. Tag value must begin and end with an alphanumeric character, and must contain only uppercase, lowercase alphanumeric characters, and the following special characters `_-.@%=+:,*#&(){}[]` and spaces.
```

Reply:
Would image-registry be degraded? I think an installer pre-check for tags' existence would be better.

Reply:
Please refer to [comment](https://issues.redhat.com/browse/CORS-2455?focusedId=23161389&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23161389). Anyway, the epic is Tech Preview in 4.14, as of now.